enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,206 @@
+        - For the time being, the MCO will have to keep some of the file writing workflow in place. In the future, plan to move these files to /usr or /etc or use systemd-tmpfiles dropins.
+
+* Questions raised alongside “moving to layered update path by default”:
+    - Q: 
+        - Should all config be built into the image? 
+        - What will the update path look like if certain config changes are not baked into the image because (1) We normally don’t write them through MachineConfig (e.g. cert writings) (2) Less reboot is favored by Node Disruption Policies?
+        - Certain config changes that the MCO watches don't trigger reboots, but it remains unknown if “moving to layered update path by default” can promise this too.
+    - A: 
+        - Everything rebootless should not be in the image, unless we have live image changes. It is better for those changes to be in writable directories on the disk. Need to keep in mind the priority order of /usr vs /etc.
+        - MCO would have to apply its own internal configs first then the MachineConfigs.
+        - [Actionable item] Get a list of files managed by the MCO to further refine this.
+        - In the future, see if bootc can handle this instead based on the decision and implementation around https://github.com/containers/bootc/issues/76.
+
+## Test Plan
+
+* Phase 1: Please refer to On Cluster Layering enhancement for more details  
+* Phase 2: TBD 
+* Phase 3 : Bootc Switch Over 
+    - Develop an e2e test suite and run payloads with bootc: Tests should target the TestOSImageURL, TestKernelArguments, TestKernelType and TestExtensions. Results will be used to discuss whether bootc is ready to perform rpm-ostree’s tasks.
+    - Create a bootc experimentation sandbox by having the BootcNodeManagement FG: When the FG is on, the rpm-ostree commands will be replaced by equivalent bootc commands to manually test the capability of bootc in non-layered machine configuration workflow. Switch back to rpm-ostree as the default option when not compatible. This experimentation prepares for Scenario 1 in the following section. 
+    - Test bootc with On Cluster Layering: When the BootcNodeManagement and OnClusterBuild FG is on, On Cluster Layering will be the default path for machine configuration, accompanied by bootc which is responsible for image roll out. This test prepares for Scenario 2 in the following section.   
+
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+There are three cases we should consider and try to support under tech preview
+  * Scenario 1: We don’t have layered update path as the default, but we start to provide rpm-ostree + bootc dual support (likely in 4.18)
+  * Scenario 2: We now have layered update path as the default and we still have rpm-ostree + bootc dual support (likely in 4.19)
+  * Scenario 3: We have completed the switch over to layered update path + bootc for image management + dnf for package management (likely in  4.20)
```

Comment:
These sound more like milestones? I'm confused by the framing as scenarios.

Reply:
These are stages we have divided the work into, so essentially milestones where we slowly add more and more support for layered update path and bootc.